docs: add components to gitbook (#569)
yetanotherco/aligned_layer: docs/architecture/components/3_service_manager_contract.md

# Aligned Service Manager Contract

> The Aligned Service Manager handles the reception of new batches to Aligned, keep their status on-chain, and receive their response.

Reviewer: keeps instead of keep
receives instead of receive

> It is a smart contract which recieves all new batches, with their merkle root and a pointer to where is the batch currently stored. When received, this manager will emit an Event for the [Operators](./4_operator.md) to know when there is a new batch to verify.

Reviewer 2: ```suggestion
It is a smart contract which receives all new batches, with their Merkle Root and a pointer to where the batch is currently stored. When received, this manager will emit an Event for the [Operators](./4_operator.md) to know when there is a new batch to verify.
```

> Then, when recieving a response from the [Aggregator](./5_aggregator.md), with Operator's aggregated BLS signatures, the Aligned Service Manager check the BLS signature to verify the Operators were in fact those who processed the batch and responded its status.

Reviewer 2: ```suggestion
Then, when receiving a response from the [Aggregator](./5_aggregator.md), with Operator's aggregated BLS signatures, the Aligned Service Manager checks the BLS signature to verify the Operators were in fact those who processed the batch and its responded status.
```

Once verified, it will emit another Event, for anyone interested (for example, the [Explorer](./6_explorer.md)) to know that the batch was verified by the operators. This batch is now verified and Users can know their proofs inside the batch were proven and verified leveraging Ethereum's security. 

## Details of the contract

> Besides the base [EigenLayer middleware contracts](https://github.com/Layr-Labs/eigenlayer-middleware/tree/mainnet/src), the core contract for Aligned is [AlignedLayerServiceManager](../../contracts/src/core/AlignedLayerServiceManager.sol). It is in charge of creating new batch verification tasks, storing batches state and verify operator responses.

Reviewer 2: ```suggestion
Besides the base [EigenLayer middleware contracts](https://github.com/Layr-Labs/eigenlayer-middleware/tree/mainnet/src), the core contract for Aligned is [AlignedLayerServiceManager](../../contracts/src/core/AlignedLayerServiceManager.sol). It is in charge of creating new batch verification tasks, storing batches state and verifying operator responses.
```

> * `batchMerkleRoot` is a 256 bit hash representing the batch merkle root the proof was included in. 

Reviewer 2: ```suggestion
* `batchMerkleRoot` is a 256 bit hash representing the batch Merkle Root the proof was included in. 
```

> * `merkleProof` is the merkle path from the hashed leaf built from the verification data commitments to the root.

Reviewer 2: ```suggestion
* `merkleProof` is the Merkle path from the hashed leaf built from the verification data commitments to the root.
```
